src/services/classifier.ts

```diff
@@ -499,8 +499,10 @@ namespace ts {
         return { spans, endOfLineState: EndOfLineState.None };
 
         function pushClassification(start: number, end: number, type: ClassificationType): void {
+            const length = end - start;
+            Debug.assert(length >= 0, `Classification had non-positive length of ${length}`);

```

Comment:
Slightly nervous about a 0-length span getting a classification unless we classify whitespace...

Reply:
Good point. It seemed like some classification spans made sense to only have trivia (since `node.getStart` is the start used here) but I now see that those won't come up in this path.